src/transport/connection/tests.rs

```diff
@@ -0,0 +1,272 @@
+use crate::client::Client;
+use crate::contracts::Contract;
+use crate::messages::{RequestHandshake, RequestMessage, ResponseMessage};
+use crate::transport::read_header;
+use log::debug;
+use std::collections::VecDeque;
+use std::io::{Read, Write};
+use std::net::TcpStream;
+use std::net::{SocketAddr, TcpListener};
+use std::str::from_utf8;
+use std::thread::JoinHandle;
+
+// Build Events with Stubs
+// struct EventBuilder {
+//     events:
+// }
+// impl EventBuilder {
+//     startup()
+// }
+
+// Trying to keep the shared functionality between the TestServer and Connection
+// in a location they can be accessed by both
+
+// #[derive(Clone, Debug)]
+// struct Message {
+//     request: RequestMessage,
+//     responses: Vec<ResponseMessage>,
+// }
+#[derive(Clone, Debug)]
+enum Message {
```

Comment:
If we treat the handshake as just `API\0` can replace this with a single struct, I like  `MockExchange`, that contains the request/replies pair.